lapce-data/src/editor.rs

```diff
@@ -288,34 +288,50 @@ impl LapceEditorBufferData {
             let path = path.clone();
             let offset = self.editor.cursor.offset();
             let prev_offset = self.doc.buffer().prev_code_boundary(offset);
-            if self.doc.code_actions.get(&prev_offset).is_none() {
-                let position = self.doc.buffer().offset_to_position(prev_offset);
-                let rev = self.doc.rev();
-                let event_sink = ctx.get_external_handle();
-                self.proxy.proxy_rpc.get_code_actions(
-                    path.clone(),
-                    position,
-                    move |result| {
-                        if let Ok(ProxyResponse::GetCodeActionsResponse {
-                            plugin_id,
-                            resp,
-                        }) = result
-                        {
-                            let _ = event_sink.submit_command(
-                                LAPCE_UI_COMMAND,
-                                LapceUICommand::UpdateCodeActions {
-                                    path,
-                                    plugin_id,
-                                    rev,
-                                    offset: prev_offset,
-                                    resp,
-                                },
-                                Target::Auto,
-                            );
-                        }
-                    },
-                );
-            }
+            // REMOVED(if) this prevents some code actions from showing up correctly, and there is almost no performance penalty for this
+            //if self.doc.code_actions.get(&prev_offset).is_none() {
```

Comment:
Does this make it not work properly?

Reply:
The if-statement causes some delayed code actions to be left out, causing some quick fixes to not appear. REMOVED(if) is just the reason why it was removed and what was removed, in this case an if statement.

Reply:
Is there any way I can test the delayed code actions bit?

Reply:
Take this rust code:

```rs
fn main() {
    let x = 0;
    x = 1;
    println!(x);
}
```

There are two errors here. Open quick fixes on the variable name, there will be no fix. Now you fix the println one, which has a higher internal priority in RA. Save the file. The bug happens when you open quick fixes on the variable name again, even after you have now fixed the println one and it shows a new error.

Reply:
The same goes for code actions on already-modified code.